### ISSUE TITLE
Only register widgets-related block in widget context (server side)

### ISSUE THREAD

**Issue by adamziel** [WordPress/gutenberg#26156]
Since legacy-widget and widget-area blocks are only registered in the widgets editor client-side, it would make sense to do the same server-side.

**Comment by draganescu** [comment]
Is [this](https://github.com/WordPress/gutenberg/pull/32501) fixing this?

**Comment by skorasaurus** [comment]
> Is [this](https://github.com/WordPress/gutenberg/pull/32501) fixing this?

maybe? with a gutenberg built from trunk earlier today; legacy widget appeared in the inserter at wp-admin/themes.php?page=gutenberg-widgets but it did not appear in the inserter when I created a new page.

**Comment by noisysocks** [comment]
No, this isn't fixed. We require the block PHP regardless of what screen is being shown.

Here in Core:

https://github.com/WordPress/wordpress-develop/blob/26722b77624fc755e3fcb721d2ce530abc777bd6/src/wp-includes/blocks/index.php#L18

Here in the plugin:

https://github.com/WordPress/gutenberg/blob/037c82baf4f8e0d4901759efed5a32f9ccd66cb6/lib/blocks.php#L174

**Comment by noisysocks** [comment]
It's a pretty minor issue as it doesn't really affect anything to my knowledge. Still, gotta shave those yaks.
